pkg/genutil/slicez/slicez.go

```diff
@@ -0,0 +1,41 @@
+package slicez
+
+// Filter iterates over elements of a slice, returning a new slice with all
+// elements that the predicate returns truthy for.
+func Filter[T any, Slice ~[]T](xs Slice, pred func(T) bool) Slice {
+	ys := make(Slice, 0, len(xs))
+	for _, x := range xs {
+		if pred(x) {
+			ys = append(ys, x)
+		}
+	}
+	return ys
+}
+
+// Map iterates over a slice and creates a new slice with each element
+// transformed.
+func Map[T any, R any](xs []T, fn func(T) R) []R {
+	ys := make([]R, len(xs))
+	for i, x := range xs {
+		ys[i] = fn(x)
```

Comment:
For my own edification: is there a functional difference between doing a set at an index vs using `append` here?

Reply:
honestly just easier on the optimizer. ideally the compiler would realize that it never has to resize when doing append for that loop and it can optimize it into the same thing as this, but there's no risk of that if you just do what i did